pkg/kbld/builder/bazel/bazel.go

```diff
@@ -0,0 +1,75 @@
+// Copyright 2020 VMware, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+package image
+
+import (
+	"bytes"
+	"fmt"
+	"io"
+	"os/exec"
+	"regexp"
+
+	ctlbdk "github.com/k14s/kbld/pkg/kbld/builder/docker"
+	"github.com/k14s/kbld/pkg/kbld/config"
+	ctllog "github.com/k14s/kbld/pkg/kbld/logger"
+)
+
+var (
+	bazelImageID = regexp.MustCompile("Loaded image ID: (sha256:)?([0-9a-z]+)")
+)
+
+type Bazel struct {
+	docker ctlbdk.Docker
+	logger ctllog.Logger
+}
+
+func NewBazel(docker ctlbdk.Docker, logger ctllog.Logger) Bazel {
+	return Bazel{docker: docker, logger: logger}
+}
+
+func (b *Bazel) Build(image, directory string, opts config.SourceBazelBuildOpts) (ctlbdk.DockerTmpRef, error) {
+
+	prefixedLogger := b.logger.NewPrefixedWriter(image + " | ")
+
+	prefixedLogger.Write([]byte(fmt.Sprintf("starting build (using bazel): %s\n", directory)))
+	defer prefixedLogger.Write([]byte("finished build (using bazel)\n"))
+
+	var imageID string
+	{
+		var stdoutBuf, stderrBuf bytes.Buffer
+
+		if opts.Label == nil {
+			return ctlbdk.DockerTmpRef{}, fmt.Errorf("Expected label to be specified, but was not")
+		}
+
+		cmdArgs := []string{"run"}
+
+		if opts.Label != nil {
```

Comment:
nit: the guard clause above makes this check technically unnecessary.

Consider pulling down that check to this point (i.e. after line 46) and removing the redundant `if`.